src/utils/streams.js

```diff
@@ -279,3 +279,35 @@ export class GunzipWrapper extends Transform {
     callback(err);
   }
 }
+
+// This tool simply forward an error happening on a stream to other streams, by
+// destroying them.
+export function forwardErrors(
```

Comment:
I didn't add unit tests for this utility but I believe the tests for the publishing cover it, by ensuring we get the right errors when feeding some bad payload to the server.